cardinal/world.go

```diff
@@ -61,11 +59,11 @@ type World struct {
 
 	// Core modules
 	worldStage       *worldstage.Manager
-	msgManager       *message.Manager
+	msgManager       MessageManager
```

Comment:
This should just be `MessageManager` similar to SystemManager. Move this and the QueryManager up to below the SystemManager